fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneSpellcheckRecordCursor.java

```diff
@@ -0,0 +1,234 @@
+        continuationBuilder.setLocation(currentPosition);
+        return ByteArrayContinuation.fromNullable(continuationBuilder.build().toByteArray());
+    }
+
+    @Override
+    public void close() {
+    }
+
+    @Nonnull
+    @Override
+    public Executor getExecutor() {
+        return executor;
+    }
+
+    @Override
+    public boolean accept(@Nonnull final RecordCursorVisitor visitor) {
+        visitor.visitEnter(this);
+        return visitor.visitLeave(this);
+    }
+
+    private synchronized IndexReader getIndexReader() throws IOException {
+        IndexWriterCommitCheckAsync writerCheck = getIndexWriterCommitCheckAsync(state, groupingKey);
+        return writerCheck == null ? DirectoryReader.open(getOrCreateDirectoryCommitCheckAsync(state, groupingKey).getDirectory()) : DirectoryReader.open(writerCheck.indexWriter);
+    }
+
+    private void spellcheck() throws IOException {
+        if (spellcheckSuggestions != null) {
+            return;
+        }
+        long startTime = System.nanoTime();
+        IndexReader indexReader = getIndexReader();
+
+        List<Suggestion> suggestionResults = new ArrayList<>();
+        for (String field : fields) {
+            //collect all suggestions across all given fields or across all fields depending on user options.
+            Arrays.stream(spellchecker.suggestSimilar(new Term(field, wordToSpellCheck), limit, indexReader))
+                    .map(suggestion -> new Suggestion(field, suggestion))
+                    .forEach(suggestionResults::add);
+        }
+        spellcheckSuggestions = suggestionResults.stream()
+                //Merge matching suggestions from different
+                .collect(Collectors.toMap(
+                        suggestion -> suggestion.suggestWord.string,
+                        Function.identity(),
+                        // TODO: For arnaud, are we checking for a merge on ALL suggested words against eachother?
+                        LuceneSpellcheckRecordCursor::mergeTwoSuggestWords))
+                .values()
+                .stream()
+                // Sort the suggested words from large to small by score then by frequency then by the field.
+                .sorted(comparing((Suggestion s) -> s.suggestWord.score).reversed()
+                                .thenComparing(comparing((Suggestion s) -> s.suggestWord.freq).reversed())
+                                .thenComparing(s -> s.suggestWord.string))
+                .limit(limit)
+                // Map the words from suggestions to index entries.
+                .map(suggestion -> new IndexEntry(
+                        state.index,
+                        Tuple.from(groupingKey.isEmpty() ? null : groupingKey.get(0).toString(), suggestion.indexField, suggestion.suggestWord.string),
+                        Tuple.from(suggestion.indexField)))
```

Comment:
Any reason the ```suggestion.indexField``` needs to occur in both the key tuple and value tuple of ```IndexEntry```?

Reply:
This was how we originally set it before incorporating your changes. I can remove it.